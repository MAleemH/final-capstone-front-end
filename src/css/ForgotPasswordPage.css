* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Montserrat', sans-serif;
  letter-spacing: 0.2px;
}

.list_style_none {
  list-style: none;
}

.td_none {
  text-decoration: none;
}

.fieldset_border_none {
  border: none;
}

.login_body {
  background-color: #fffcf6;
}

.login_header {
  width: 90%;
  margin: auto;
  padding: 10px 0;
  height: 40px;
  position: sticky;
  top: 0;
  left: 0;
  background: transparent;
}

.login_header img {
  height: 30px;
}

.login_main {
  width: 40%;
  margin: auto;
}

.logo_fig {
  display: flex;
  justify-content: center;
  padding: 65px 0;
}

.eyebox img {
  width: 30px;
}

.login_main .logo_fig img {
  height: 100px;
  margin: auto;
}

.fadebreak_box {
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
}

.login_form {
  padding: 20px 0;
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  gap: 15px;
}

.login_form fieldset {
  display: flex;
  flex-direction: column;
  gap: 15px;
  width: 100%;
}

.login_form .input_name {
  width: 100%;
  color: #172b4d;
  font-size: 15px;
  padding: 15px;
  outline: none;
  border: 1px solid #d9e1f6;
  border-radius: 10px;
}

.password_box {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid #d9e1f6;
  border-radius: 10px;
  padding-right: 5px;
}

.login_form .input_password {
  width: 100%;
  color: #172b4d;
  font-size: 15px;
  padding: 15px;
  outline: none;
  border: none;
  border-radius: 10px;
}

.login_form input::placeholder {
  color: #172b4d;
  font-size: 17px;
  font-weight: 400;
  line-height: 28px;
  transition: 3s;
}

.login_form input:hover::placeholder {
  transform: translate(10px, 0);
  color: #172b4d50;
}

.login_action {
  width: 20%;
  margin: auto;
}

.login_action button {
  padding: 10px 8px;
  border-radius: 10px;
  background-color: #ff9186;
  border: 1px solid #d9e1f6;
  color: #172b4d;
}

.login_main .login_main_div {
  display: flex;
  justify-content: end;
  gap: 20px;
  align-items: center;
}

.login_footer ul a {
  color: #172b4d;
}

.login_main .login_main_div a {
  color: #172b4d;
}

.login_main .login_main_div button {
  color: #172b4d;
  padding: 10px 8px;
  border-radius: 10px;
  background-color: #ff9186;
  border: 1px solid #d9e1f6;
}

.login_footer {
  width: 90%;
  margin: auto;
  position: sticky;
  bottom: 0;
  left: 0;
  background: transparent;
  padding: 10px 0;
}

.login_footer ul {
  display: flex;
  gap: 15px;
}

.password_display {
  border: none;
  background-color: transparent;
}

.rotate_breakline {
  transform: rotate(180deg);
}
