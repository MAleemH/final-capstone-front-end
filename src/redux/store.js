import { configureStore } from '@reduxjs/toolkit';
import therapyReducer from './Therapy/therapySlice';
import userReducer from './User/userSlice';

const store = configureStore({
  reducer: {
    therapy: therapyReducer,
    user: userReducer,
  },
});

export default store;
