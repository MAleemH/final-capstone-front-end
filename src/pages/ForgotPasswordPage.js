import React, { useEffect, useState } from 'react';
import { Link, useNavigate } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { forgotPassword } from '../redux/User/userSlice';
import '../css/ForgotPasswordPage.css';
import backImg from '../img/back.png';
import eyeImg from '../img/eye.png';
import fadingBreak from '../img/fading_break.png';
import logoImg from '../img/logo.png';

function ForgotPasswordPage() {
  const dispatch = useDispatch();
  const [email, setEmail] = useState('');
  const [inputPas1, setinputPas1] = useState('');
  const [inputPas2, setinputPas2] = useState('');
  const [errorMessage, setErrorMessage] = useState('');
  const [passwordType1, setPasswordType1] = useState('password');
  const [passwordType2, setPasswordType2] = useState('password');
  const navigate = useNavigate();

  const goBack = () => {
    navigate(-1);
  };

  const similarPassword = async (inputPas1, inputPas2) => {
    if (inputPas1 !== inputPas2) {
      setErrorMessage('Unmatch Pass');
      return;
    }
    setErrorMessage('');
  };

  const nullUserData = async () => {
    setEmail('');
    setinputPas1('');
    setinputPas2('');
  };

  const handleForgotPass = async () => {
    const pass1 = inputPas1.replace(/\s/g, '').toLowerCase();
    const pass2 = inputPas2.replace(/\s/g, '').toLowerCase();
    console.log(email, inputPas1, inputPas2);
    await similarPassword(pass1, pass2);
    const userData = {
      user: {
        password: inputPas1, password_confirmation: inputPas2, email,
      },
    };

    console.log(userData);
    await dispatch(forgotPassword(userData));
    await nullUserData();
  };

  useEffect(() => {
    let active = true;
    (async () => {
      if (active) {
        setErrorMessage('');
      }
    })();
    return () => {
      active = false;
    };
  }, []);

  return (
    <div className="forgot_page_body">

      <header className="forgot_page_header">
        <nav>
        <button type="button" onClick={goBack}>
            <img src={backImg} alt="" />
          </button>
        </nav>
      </header>

      <main className="forgot_page_main">
        <figure className="logo_fig">
          <Link to="/">
            <img src={logoImg} alt="" />
          </Link>
        </figure>

        <div className="fadebreak_box">
          <figure>
            {' '}
            <img src={fadingBreak} alt="" />
            {' '}
          </figure>
          <p>
&nbsp;
            {`${errorMessage || 'Forgot Password'}`}
          </p>
          <figure>
            {' '}
            <img className="rotate_breakline" src={fadingBreak} alt="" />
            {' '}
          </figure>
        </div>

        <form action="" className="forgot_page_form">
          <fieldset className="fieldset_border_none">
            <input className="input_name" type="email" value={email} placeholder="Email" aria-label="Input Email" onChange={(e) => setEmail(e.target.value)} required />

            <div className="password_box">
              <input className="input_password" type={passwordType1} value={inputPas1} placeholder="Password" aria-label="Input Password" onChange={(e) => setinputPas1(e.target.value)} required />
              {passwordType1 === 'password'
                ? (
                  <figure className="eyebox">
                    <button className="password_display" type="button" onClick={() => setPasswordType1('text')}>
                      <img src={eyeImg} alt="" />
                    </button>
                  </figure>
                )
                : (
                  <figure className="eyebox">
                    <button className="password_display" type="button" onClick={() => setPasswordType1('password')}>
                      <img src={eyeImg} alt="" />
                    </button>
                  </figure>
                )}
            </div>

            <div className="password_box">
              <input className="input_password" type={passwordType2} value={inputPas2} placeholder="Re-enter Password" aria-label="Input Password" onChange={(e) => setinputPas2(e.target.value)} required />
              {passwordType2 === 'password'
                ? (
                  <figure className="eyebox">
                    <button className="password_display" type="button" onClick={() => setPasswordType2('text')}>
                      <img src={eyeImg} alt="" />
                    </button>
                  </figure>
                )
                : (
                  <figure className="eyebox">
                    <button className="password_display" type="button" onClick={() => setPasswordType2('password')}>
                      <img src={eyeImg} alt="" />
                    </button>
                  </figure>
                )}
            </div>
          </fieldset>

          <fieldset className="fieldset_border_none forgot_page_action">
            <button type="button" onClick={handleForgotPass}>Submit</button>
          </fieldset>
        </form>

        <div className="forgot_page_main_div">
          <p>
            {' '}
            <Link className="td_none" to="/register"> need an account? </Link>
            {' '}
          </p>

          <button type="button">
            <Link className="td_none" to="/forgot_page"> LOGIN </Link>
          </button>
        </div>

      </main>

      <footer className="forgot_page_footer">
        <ul className="list_style_none">
          <li>
            {' '}
            <Link className="td_none" to="/register"> terms </Link>
            {' '}
          </li>
          <li>
            {' '}
            <Link className="td_none" to="/register"> privacy </Link>
            {' '}
          </li>
        </ul>
      </footer>

    </div>
  );
}

export default ForgotPasswordPage;
