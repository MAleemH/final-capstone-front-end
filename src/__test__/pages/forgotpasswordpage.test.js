import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import ForgotPasswordPage from '../../pages/ForgotPasswordPage';

it('ForgotPasswordPage component renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <ForgotPasswordPage />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
