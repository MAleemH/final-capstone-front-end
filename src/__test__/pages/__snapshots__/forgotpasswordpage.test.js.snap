// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordPage component renders correctly 1`] = `
<div
  className="forgot_page_body"
>
  <header
    className="forgot_page_header"
  >
    <nav>
      <button
        className="back_none"
        onClick={[Function]}
        type="button"
      >
        <img
          alt=""
          src="back.png"
        />
      </button>
    </nav>
  </header>
  <main
    className="forgot_page_main"
  >
    <figure
      className="logo_fig"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt=""
          src="logo.png"
        />
      </a>
    </figure>
    <div
      className="fadebreak_box"
    >
      <figure>
         
        <img
          alt=""
          src="fading_break.png"
        />
         
      </figure>
      <p>
        Â 
        Forgot Password
      </p>
      <figure>
         
        <img
          alt=""
          className="rotate_breakline"
          src="fading_break.png"
        />
         
      </figure>
    </div>
    <form
      action=""
      className="forgot_page_form"
    >
      <fieldset
        className="fieldset_border_none"
      >
        <input
          aria-label="Input Email"
          className="input_name"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="email"
          value=""
        />
        <div
          className="password_box"
        >
          <input
            aria-label="Input Password"
            className="input_password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value=""
          />
          <figure
            className="eyebox"
          >
            <button
              className="password_display"
              onClick={[Function]}
              type="button"
            >
              <img
                alt=""
                src="eye.png"
              />
            </button>
          </figure>
        </div>
        <div
          className="password_box"
        >
          <input
            aria-label="Input Password"
            className="input_password"
            onChange={[Function]}
            placeholder="Re-enter Password"
            required={true}
            type="password"
            value=""
          />
          <figure
            className="eyebox"
          >
            <button
              className="password_display"
              onClick={[Function]}
              type="button"
            >
              <img
                alt=""
                src="eye.png"
              />
            </button>
          </figure>
        </div>
      </fieldset>
      <fieldset
        className="fieldset_border_none forgot_page_action"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Submit
        </button>
      </fieldset>
    </form>
    <div
      className="forgot_page_main_div"
    >
      <p>
         
        <a
          className="td_none"
          href="/register"
          onClick={[Function]}
        >
           need an account? 
        </a>
         
      </p>
      <button
        type="button"
      >
        <a
          className="td_none"
          href="/login"
          onClick={[Function]}
        >
           LOGIN 
        </a>
      </button>
    </div>
  </main>
  <footer
    className="forgot_page_footer"
  >
    <ul
      className="list_style_none"
    >
      <li>
         
        <a
          className="td_none"
          href="/register"
          onClick={[Function]}
        >
           terms 
        </a>
         
      </li>
      <li>
         
        <a
          className="td_none"
          href="/register"
          onClick={[Function]}
        >
           privacy 
        </a>
         
      </li>
    </ul>
  </footer>
</div>
`;
